core_impl/src/main/java/io/opencensus/implcore/tags/TaggerImpl.java

```diff
@@ -52,16 +62,19 @@ public TagContextBuilder currentBuilder() {
   }
 
   @Override
-  public TagContextBuilderImpl toBuilder(TagContext tags) {
-    return toTagContextBuilderImpl(tags);
+  public TagContextBuilder toBuilder(TagContext tags) {
+    return toTagContextBuilderImpl(tags, state.get());
   }
 
   @Override
   public Scope withTagContext(TagContext tags) {
-    return CurrentTagContextUtils.withTagContext(toTagContextImpl(tags));
+    return CurrentTagContextUtils.withTagContext(toTagContextImpl(tags, state.get()));
   }
 
-  private static TagContextImpl toTagContextImpl(TagContext tags) {
+  private static TagContextImpl toTagContextImpl(TagContext tags, TaggingState state) {
```

Comment:
I would rather make this non-static, up to you.

Reply:
I meant because you passed an internal state (the state) as argument to keep this static instead of making it instance method.

Reply:
Now I see what you mean.  I passed in the state enum (instead of giving access to the AtomicReference) to make it clear that the method only needs a snapshot of the state and doesn't need to update it.